src/main/java/io/pravega/connectors/hadoop/AbstractPravegaBuilder.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.pravega.connectors.hadoop;
+
+import org.apache.hadoop.conf.Configuration;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * An abstract pravega hadoop connector builder.
+ *
+ * The builder is abstracted to act as the base for both the {@link PravegaInputFormatBuilder} builder.
+ *
+ * @param <B> the builder type.
+ */
+abstract class PravegaBuilder<B extends PravegaBuilder> {
```

Comment:
I agree with having a base builder class for dealing with settings that would apply to both `PravegaInputFormat` and `PravegaOutputFormat`.    But I see only `withURI` as a setting that would apply to both.    I would say, the others need only be defined on `PravegaInputFormatBuilder`.

Reply:
accepted. only keep URI, scope, and stream which apply both input and output formats.